modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractCSharpCodegen.java

```diff
@@ -307,10 +307,26 @@ public void processOpts() {
                 }
             }
         }
+
+        processUsings(objs);
+
         // process enum in models
         return postProcessModelsEnum(objs);
     }
 
+    private void processUsings(Map<String, Object> objs) {
```

Comment:
I think I get what you're saying.

It's not really possible to prevent developers from misusing the templates (removing the fully qualified model references). Your last comment about the templates knowing the difference between Model references and core references can easily be done in the generator's code when parameters are being mapped, because we know which parameters come from the swagger definition (they point to a `ModelRef`, if I remember correct). I know @wing328 was working on a solution to fully qualify models but I don't know the state of that work.